### PR TITLE
Issue #11791 fix suffix mapping for non default DefaultServlet usage.

### LINKED ISSUE #11791: How to configure Jetty Server v12 ee10 for serving multiple static resources in the same base directory?
**Jetty version(s)**
12.0.9

**Jetty Environment**
ee10

**Java Version**
JavaSE-17

I am working on a web application using Java, and I have configured servlets to serve static resources using ServletHolder and DefaultServlet. Here is the configuration code:

`ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);`
`context.setContextPath("/");`
`ResourceFactory resourceFactory = ResourceFactory.of(context);`
`context.setBaseResource(resourceFactory.newResource(mainResourceBase));`
`ServletHolder holderAlt = new ServletHolder("static-alt", DefaultServlet.class);`
`holderAlt.setInitParameter("dirAllowed", "true");`
` holderAlt.setInitParameter("acceptRanges", "true");`
`context.addServlet(holderAlt, "*.js");`
`ServletHolder holderDef = new ServletHolder("default", DefaultServlet.class);`
`holderDef.setInitParameter("dirAllowed", "true");`
`context.addServlet(holderDef, "/");`

i'm encountering a 404 Not Found error when trying to access static resources ending with .js.
I would appreciate any insights or suggestions to help resolve the issue of static resources returning 404 errors.

### PR DESCRIPTION
**Pull request by janbartel**
Closes #11791

### CODE REVIEW

File: jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java [hunk 1]
```diff
@@ -555,7 +556,20 @@ protected String getEncodedPathInContext(HttpServletRequest req, String included
...
+            String path = req.getPathInfo();
+            if (MappingMatch.EXTENSION.equals(req.getHttpServletMapping().getMappingMatch()))
+            {
+                if (StringUtil.isEmpty(path))
+                    return encodePath(req.getServletPath());
+                else
+                    return encodePath(path);
+            }
+
+            return encodePath(path);
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java]:
```suggestion
            String path = req.getPathInfo();
            if (StringUtil.isEmpty(path) && 
                MappingMatch.EXTENSION.equals(req.getHttpServletMapping().getMappingMatch()))
                path = req.getServletPath();
          
            return encodePath(path);
```

**Review by gregw — CHANGES_REQUESTED** [review]

**Review by lorban — APPROVED** [review]

**Review by gregw — APPROVED** [review]

**Comment by Emilyserap** [comment]
when this pull request will be merge ?

**Comment by Emilyserap** [comment]
Thanks for fixing this issue, but I want to know why when I refresh the dependencies, I don't see the new modification in the class DefaultServlet.java

**Comment by lachlan-roberts** [comment]
@Emilyserap this is assigned to the `12.0.10` project, so you will need to wait for the release (which should be later this week), or you can build from the current `12.0.x` branch yourself.

**Pull request merged**